### ISSUE TITLE
Linter fails in mac with failed to load package message

### ISSUE THREAD

**Issue by subusc** [golangci/golangci-lint#3391]
### Welcome

- [X] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [X] Yes, I've searched similar issues on GitHub and didn't find any.
- [X] Yes, I've included all information below (version, config, etc.).
- [X] Yes, I've tried with the standalone linter if available (e.g., gocritic, go vet, etc.). (https://golangci-lint.run/usage/linters/)

### Description of the problem

golangci-lint-mod........................................................Failed
- hook id: golangci-lint-mod
- exit code: 1

WARN [runner] Can't run linter goanalysis_metalinter: inspect: failed to load package rados: could not load export data: no export data for "github.com/ceph/go-ceph/rados"
ERRO Running error: 1 error occurred:
	* can't run linter goanalysis_metalinter: inspect: failed to load package rados: could not load export data: no export data for "github.com/ceph/go-ceph/rados"

### Version of golangci-lint

<details>

```console
 % golangci-lint --version
golangci-lint has version v1.50.1 built from (unknown, mod sum: "h1:C829clMcZXEORakZlwpk7M4iDw2XiwxxKaG504SZ9zY=") on (unknown)
```

</details>

### Configuration file

<details>

```console % 
cat .golangci-lint.yaml
linters:
  enable:
    - asasalint # checks for pass []any as any in variadic func(...any)
    - asciicheck # checks that your code does not contain non-ASCII identifiers
    - bidichk # checks for dangerous unicode character sequences
    - bodyclose # checks whether HTTP response body is closed successfully
    - cyclop # checks function and package cyclomatic complexity
    - dupl # tool for code clone detection
    - durationcheck # checks for two durations multiplied together
    - errname # checks that sentinel errors are prefixed with the Err and error types are suffixed with the Error
    - errorlint # finds code that will cause problems with the error wrapping scheme introduced in Go 1.13
    - execinquery # checks query string in Query function which reads your Go src files and warning it finds
    - exhaustive # checks exhaustiveness of enum switch statements
    - exportloopref # checks for pointers to enclosing loop variables
    - forbidigo # forbids identifiers
    - funlen # tool for detection of long functions
    - gochecknoinits # checks that no init functions are present in Go code
    - gocognit # computes and checks the cognitive complexity of functions
    - goconst # finds repeated strings that could be replaced by a constant
    - gocritic # provides diagnostics that check for bugs, performance and style issues
    - gocyclo # computes and checks the cyclomatic complexity of functions
    - goimports # in addition to fixing imports, goimports also formats your code in the same style as gofmt
    - gomodguard # allow and block lists linter for direct Go module dependencies. This is different from depguard where there are different block types for example version constraints and module recommendations
    - goprintffuncname # checks that printf-like functions are named with f at the end
    - gosec # inspects source code for security problems
    - loggercheck # checks key value pairs for common logger libraries (kitlog,klog,logr,zap)
    - makezero # finds slice declarations with non-zero initial length
    - misspell # finds commonly misspelled English words in comments
    - nakedret # finds naked returns in functions greater than a specified function length
    - nestif # reports deeply nested if statements
    - nilerr # finds the code that returns nil even if it checks that the error is not nil
    - noctx # finds sending http request without context.Context
    - nolintlint # reports ill-formed or insufficient nolint directives
    - nonamedreturns # reports all named returns
    - predeclared # finds code that shadows one of Go's predeclared identifiers
    - promlinter # checks Prometheus metrics naming via promlint
    - reassign # checks that package variables are not reassigned
    - revive # fast, configurable, extensible, flexible, and beautiful linter for Go, drop-in replacement of golint
    - tenv # detects using os.Setenv instead of t.Setenv since Go1.17
    - testableexamples # checks if examples are testable (have an expected output)
    - tparallel # detects inappropriate usage of t.Parallel() method in your Go test codes
    - unconvert # removes unnecessary type conversions
    - unparam # reports unused function parameters
    - usestdlibvars # detects the possibility to use variables/constants from the Go standard library
    - whitespace # detects leading and trailing whitespace

linters-settings:
  cyclop:
    max-complexity: 30
    package-average: 10.0
  gci:
    sections:
      - standard
      - default
      - prefix(github.sc-corp.net)
  exhaustive:
    check:
      - switch
      - map
  funlen:
    lines: 100
    statements: 50
  goimports:
    local-prefixes: github.sc-corp.net
  govet:
    check-shadowing: true
    enable-all: true
    disable:
      - fieldalignment
  misspell:
    locale: US
  nolintlint:
    allow-no-explanation: [ funlen, gocognit ]
    require-explanation: true
    require-specific: true
  revive:
    rules:
      - name: var-naming
        arguments: [["URL", "CDN", "API"], []]
  # block any version of local modules to force replace directives
  forbidigo:
    forbid:
      - '(?i)whitelist'
      - '(?i)blacklist'
      - '(?i)master'
      - '(?i)slave'
  tenv:
    all: true
  gosec:
    excludes:
      - G402
  nestif:
    min-complexity: 8

run:
  timeout: 3m
  parallel: true


issues:
  exclude-rules:
    - source: "(noinspection|TODO)"
      linters: [ godot ]
    - source: "//noinspection"
      linters: [ gocritic ]
    - source: "^\\s+if _, ok := err\\.\\([^.]+\\.InternalError\\); ok {"
      linters: [ errorlint ]
    - path: "_test\\.go"
      linters:
        - bodyclose
        - dupl
        - funlen
        - goconst
        - gosec
        - noctx
        - wrapcheck
        - forbidigo
```

</details>

### Go environment

<details>

```console 
% go version && go env
go version go1.19 darwin/amd64
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/Users/subramanian.sankarasubramanian/Library/Caches/go-build"
GOENV="/Users/subramanian.sankarasubramanian/Library/Application Support/go/env"
GOEXE=""
GOEXPERIMENT=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOINSECURE=""
GOMODCACHE="/Users/subramanian.sankarasubramanian/Snapchat/Dev/go/pkg/mod"
GONOPROXY="*.sc-corp.net"
GONOSUMDB="*.sc-corp.net"
GOOS="darwin"
GOPATH="/Users/subramanian.sankarasubramanian/Snapchat/Dev/go"
GOPRIVATE="*.sc-corp.net"
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/Users/subramanian.sankarasubramanian/Snapchat/Dev/.asdf/installs/golang/1.19/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/Users/subramanian.sankarasubramanian/Snapchat/Dev/.asdf/installs/golang/1.19/go/pkg/tool/darwin_amd64"
GOVCS=""
GOVERSION="go1.19"
GCCGO="gccgo"
GOAMD64="v1"
AR="ar"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
GOMOD="/Users/subramanian.sankarasubramanian/Snapchat/workspace/go/src/github.sc-corp.net/Snapchat/S4/go.mod"
GOWORK=""
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -arch x86_64 -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/xy/pc8y55jj6qb3k_yxflwzgm440000gq/T/go-build1613793457=/tmp/go-build -gno-record-gcc-switches -fno-common"
```

</details>

### Verbose output of running

<details>

```console
 % golangci-lint run -v
INFO [config_reader] Config search paths: [./ /Users/subramanian.sankarasubramanian/Snapchat/workspace/go/src/github.sc-corp.net/Snapchat/S4 /Users/subramanian.sankarasubramanian/Snapchat/workspace/go/src/github.sc-corp.net/Snapchat /Users/subramanian.sankarasubramanian/Snapchat/workspace/go/src/github.sc-corp.net /Users/subramanian.sankarasubramanian/Snapchat/workspace/go/src /Users/subramanian.sankarasubramanian/Snapchat/workspace/go /Users/subramanian.sankarasubramanian/Snapchat/workspace /Users/subramanian.sankarasubramanian/Snapchat /Users/subramanian.sankarasubramanian /Users /]
INFO [lintersdb] Active 7 linters: [errcheck gosimple govet ineffassign staticcheck typecheck unused]
INFO [loader] Go packages loading at mode 575 (files|imports|name|compiled_files|exports_file|types_sizes|deps) took 2.56629198s
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 5.689009ms
INFO [linters_context/goanalysis] analyzers took 3m11.046956867s with top 10 stages: buildir: 2m14.758990861s, nilness: 7.278215767s, printf: 6.64208038s, fact_deprecated: 6.323453047s, ctrlflow: 6.09603322s, fact_purity: 5.154907916s, typedness: 4.830259371s, SA5012: 4.335936428s, inspect: 3.865657385s, SA4030: 289.939703ms
WARN [runner] Can't run linter goanalysis_metalinter: buildir: failed to load package rados: could not load export data: no export data for "github.com/ceph/go-ceph/rados"
INFO [runner] processing took 6.865µs with stages: max_same_issues: 2.502µs, nolint: 2.247µs, skip_dirs: 293ns, max_from_linter: 236ns, uniq_by_line: 158ns, cgo: 157ns, autogenerated_exclude: 145ns, filename_unadjuster: 131ns, skip_files: 130ns, path_prettifier: 128ns, identifier_marker: 121ns, exclude: 120ns, path_shortener: 69ns, sort_results: 67ns, diff: 66ns, source_code: 62ns, max_per_file_from_linter: 61ns, severity-rules: 59ns, exclude-rules: 57ns, path_prefixer: 56ns
INFO [runner] linters took 15.456662043s with stages: goanalysis_metalinter: 15.456608611s
ERRO Running error: 1 error occurred:
	* can't run linter goanalysis_metalinter: buildir: failed to load package rados: could not load export data: no export data for "github.com/ceph/go-ceph/rados"

INFO Memory: 176 samples, avg is 1886.6MB, max is 3420.5MB
INFO Execution took 18.325731055s
```

</details>

### Code example or link to a public repository

<details>

```go

package main

import (
        "github.com/ceph/go-ceph/rados" //nolint
)

func main() {
	_,_ := rados.NewConn()
}
```

</details>

**Comment by bombsimon** [comment]
Did you ensure building works for you before you try to lint the code? The error indicates problems before the analysis step and if I try to build your code I get the following error:

```sh
$ go build ./
# github.com/ceph/go-ceph/rados
../../go/pkg/mod/github.com/ceph/go-ceph@v0.18.0/rados/command.go:5:11: fatal error: 'rados/librados.h' file not found
 #include <rados/librados.h>
          ^~~~~~~~~~~~~~~~~~
1 error generated.
```

**Comment by subusc** [comment]
Thanks a ton for the quick reply - I have noticed this elsewhere as well: Build breaks/warnings do block golangci - but the error messages aren't directly indicative. Is it possible to update the error messages to ensure things build first?

FWIW this was blocked by https://github.com/ceph/go-ceph/issues/794 - which I am following up on! Thanks!

**Comment by gabizou** [comment]
This unfortunately isn't down to building but rather a result of the go-ast resolver. If a package has a dependency on another package that has links for C code that cannot be resolved, the entire package is deemed an error/unresolvable (this is performed not by golangci-lint itself, but the [golang/x/tools/go/analysis analyzer](https://cs.opensource.google/go/x/tools/+/master:go/analysis/analysis.go). Because the errors end up coming from deep within the tool, it's not something exposed back except to say "hey, this package can't be resolved".

As previously commented, the c-library code dependency is required for the package to be resolved, then golangci-lint (analysis) will be able to deem the imported package as resolved.
